### ISSUE TITLE
[Effect Suggestion] Lock y position

### ISSUE THREAD

**Issue by anto4440** [gta-chaos-mod/ChaosModV#980]
**Name of effect**
Lock 'y' position

**Duration if timed effect (long or short timed?)**
Short

**Effect Description**
Locks players 'y' position. If they can still move around. If there is a hill, it will act like a wall. If the player walks off a hill, instead of falling they will simply walk in the air.

**Comment by Last0xygen** [comment]
I think you mean Z-position ;)

![](https://i.ytimg.com/vi/VA5AmjhTA3A/maxresdefault.jpg)

**Comment by Last0xygen** [comment]
I think you mean „z-position“ ;)

![](https://i.ytimg.com/vi/VA5AmjhTA3A/maxresdefault.jpg)

But also: this would result in killing the player if he is driving around as it would crush him on every little bump. No road is completely flat :/

**Comment by anto4440** [comment]
No, because you wont take damage from a wall. Maybe even for a split second give an antiforcefeild so that player doesn't immediately die from a bump.

**Comment by Last0xygen** [comment]
If the player rides a bike with high speed though? 🤔😅 making a jump or driving downhill would result in „driving against a wall“, right?

**Comment by anto4440** [comment]
Making a jump would have very different results:
Option a: Player is on a flat plane, nearing a jump. Player slams into ramp, maybe gets thrown off the bike, and that's it.
Option b: Player is mid-ramp when effect is applied. Same thing happens.
Option c: Player is in the air when effect is applied. Well, the player can now float and fly around until effect ends.

And if the player is going downhill... He floats until effect ends.

**Comment by TheRealAyCe** [comment]
3D games/engines are not known to be consistent for assigning Y or Z to up/down. :P
